woocommerce/woocommerce
# [LYS] Try single coming soon template

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR investigates using a single template to render instead of a Coming Soon page itself. Patterns are used to populate the template dynamically based on the `woocommerce_store_pages_only` option value.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Fire up a JN instance with `launch-your-store` enabled
2. Check out the `Page: Coming soon` template in the editor Appearance > Editor > Templates > Page: Coming soon. Note https://github.com/woocommerce/woocommerce/issues/46355 has not yet been addressed.
3. In a separate tab switch the Site Visibility setting "Restrict to store pages only" at `/wp-admin/admin.php?page=wc-settings&tab=site-visibility`
4. Refresh the Site Editor with the template, see the content and layout change
5. Depending on your setting, visit a front end page using incognito mode to make sure the routing works
6. Switch the Site Visibilty setting again and check out a corresponding page in incognito mode to see the content appropriately change



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

Change to unreleased feature

</details>
